### PR TITLE
Week1 -level 1, 2

### PR DESCRIPTION
**Pull request by AnilKumarN**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,4 +77,23 @@ At the command line, do the following steps to run the example.  For purposes of
...
+1. `pip install` all of the libraries you see in `.gitpod.Dockerfile`
+1. Setup your weekly python environments per the "Weekly Project" above.
+1. Install [Logstash with the OpenSearch Plugin](https://opensearch.org/docs/latest/clients/logstash/index/)
+1. Install [Fasttext](https://fasttext.cc/)  
+1. Run OpenSearch: 
+    1. `cd docker`
+    1. `docker-compose up`
+1. Note: most of the scripts and projects assume the data is in `/workspace/datasets`, but have overrides to specify your own directories. You will need to download and plan accordingly.  
+1. Do your work per the Weekly Project     
+
```
Comment (andreportaro) [inline README.md:99]: Very nice! If you haven't you could also create a PR just with this commit and pointing to the main repo

File: opensearch/bbuy_products.json [hunk 2]
```diff
@@ -1,5 +1,462 @@
...
+        "type": "text",
+        "fields": {
+          "keyword": {
+            "type": "keyword",
+            "ignore_above": 256
+          }
+        }
+      },
+      "shortDescription": {
+        "type": "text",
```
Comment (andreportaro) [inline opensearch/bbuy_products.json:368]: just a nitpick:  if you do not use an analyzer on the `name`, `shortDescription` and `longDescription`, then [the whole text is treated as a keyword](https://codecurated.com/blog/elasticsearch-text-vs-keyword/), meaning that you cannot search "within" the text - you can only do exact matches

File: opensearch/bbuy_products.json [hunk 3]
```diff
@@ -1,5 +1,462 @@
...
+      "modelNumber": {
+        "type": "text",
+        "fields": {
+          "keyword": {
+            "type": "keyword",
+            "ignore_above": 256
+          }
+        }
+      },
+      "name": {
```
Comment (andreportaro) [inline opensearch/bbuy_products.json:277]: Same comment as below - adding an `english` analyzer to enable searching within a field, or removing stopwords etc might be benefitial

File: opensearch/bbuy_queries.json [hunk 4]
```diff
@@ -1,4 +1,32 @@
...
+        "type": "text",
+        "fields": {
+          "keyword": {
+            "type": "keyword"
+          }
+        },
+        "analyzer": "english"
+      },
+      "click_time": {
+        "type": "keyword"
```
Comment (andreportaro) [inline opensearch/bbuy_queries.json:23]: Having `keyword` type analyzers for date fields will not allow using them for more advanced date queries such as range etc.

File: week1/search.py [hunk 5]
```diff
@@ -88,13 +124,109 @@ def query():
...
+    multi_match_query_obj = {
+        "multi_match": {
+            "query": user_query,
+            "fields": ["name^100", "shortDescription^50", "longDescription^20", "department^10"]
+        }
+    }
+
+    if user_query == '*':
+        multi_match_query_obj = {
+            "match_all": {}
```
Comment (andreportaro) [inline week1/search.py:136]: the `query_string` query allows the `*` wildcard so this wouldn't be 100% necessary
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#query-string-top-level-params

File: week3/conf/ltr_featureset.json [hunk 6]
```diff
@@ -0,0 +1,327 @@
+{
```
Comment (andreportaro) [inline week3/conf/ltr_featureset.json:1]: Wow, looks like you're already pretty advanced! Good one!

**Review by andreportaro — COMMENTED** [review]
Nice work!!

File: README.md (hunk 1 above)
Comment (AnilKumarN) [reply README.md:99]: Yes. I forgot to take the latest pull. And some changes was done to the main repo. Will keep this in mind

File: opensearch/bbuy_products.json (hunk 2 above)
Comment (AnilKumarN) [reply opensearch/bbuy_products.json:368]: Yes

File: opensearch/bbuy_products.json (hunk 3 above)
Comment (AnilKumarN) [reply opensearch/bbuy_products.json:277]: Yes

File: week1/search.py (hunk 5 above)
Comment (AnilKumarN) [reply week1/search.py:136]: ok

File: week3/conf/ltr_featureset.json (hunk 6 above)
Comment (AnilKumarN) [reply week3/conf/ltr_featureset.json:1]: These are some changes pushed by **gsingers** to the main repo..My pull request has incorporated those too

**Pull request merged**